api/src/main/scala/org/ensime/api/outgoing.scala

```diff
@@ -177,6 +187,8 @@ case class SourceFileInfo(
 ) {
   // keep the log file sane for unsaved files
   override def toString = s"SourceFileInfo($file,${contents.map(_ => "...")},$contentsIn)"
+  def isJava = file.getName().endsWith(".java")
```

Comment:
I'd prefer if we moved any code from the API module into somewhere else (the typeclass approach vs OOP) as it keeps the API schema definition really clean. (it's not possible to use typeclasses for toString, equals, hashCode unfortunately)